File: src/main/kotlin/poo/Construtores.java
package poo;

public class Construtores {

    private String nome;
    private int anoNascimento;

    // constructor
    public Construtores(String nome, int anoNascimento) {
        this.nome = nome;
        this.anoNascimento = anoNascimento;
    }

    public static void main(String[] args) {
        Construtores p = new Construtores("Jose", 1999);
        System.out.println(p.nome);
    }

}

File: src/main/kotlin/poo/construtores/Construtores.kt
package poo.construtores

class Pessoa2(var nome: String){

    // liberando acesso ao attr ano
    var ano: Int? = null

    // construtur segudário
    // obrigatório a chamada do construtor primário
    // acresentar mais atributos para diferenciar do primário
    // os atributos do contruturo são acessiveis em outro local
    constructor(nome: String, ano: Int) : this(nome){
        this.ano = ano // acessando o attr do construtor
    }

    fun saudacao(){
        println("Olá, meu nome é $nome e nasci no ano $ano")
    }

}

fun main() {

    val p1: Pessoa2 = Pessoa2("Fulano")
    val p2: Pessoa2 = Pessoa2("Silva", 1990)
    p2.saudacao()
    p1.saudacao()

}